(react-switch) - Adding Spec.md (#19426)
microsoft/fluentui: packages/react-switch/Spec.md

| styles       | Optional styles for the component.                                                              | Not used in converged components                   |
| role         | Whether to use the 'switch' role (ARIA 1.1) or the 'checkbox' role (ARIA 1.0).                  | Toggle has a hidden input element.                 |

## Structure

- _**Public**_

```jsx=
<Switch checked={true}/>
```

- _**Internal**_

```jsx=
<slots.root {...slotProps.root}>
  <slots.rail {...slotProps.rail} />
  <slots.track {...slotProps.track} />
  <slots.thumbWrapper {...slotProps.thumbWrapper}>
    <slots.thumb {...slotProps.thumb} />
  </slots.thumbWrapper>
  <input type="checkbox" />
</slots.root>;
```

- \_**DOM**

```jsx=
<div className="ms-switch-root">
  <div className="ms-switch-rail" />
  <div className="ms-switch-track" />
  <div className="ms-switch-thumbWrapper">
    <div className="ms-switch-thumb" />
  </div>
  <input type="checkbox" />
</div>;
```

## Behaviors

### Component States

- **Disabled**
  - When disabled, all events are ignored, and the Switch's value never updates.

>   - Does not allow focus.

Reviewer: You should probably mention cursor changes to `not-allowed` for the disabled state.

- **Checked**

>   - Toggle is off when the “thumb” is indicated on the left.
>   - Toggle is on when the thumb is indicated on the right.

Reviewer: nit:

```suggestion
  - Switch is off when the “thumb” is indicated on the left.
  - Switch is on when the "thumb" is indicated on the right.
```

  - This is switched in RTL.
  - When switching toggles on and off, the on state should change using the checked state appearance styles.

### Hover

> The cursor changes to the hand icon. The outline of the toggle and thumb should also darken. This helps reinforce that the area is interactive.

Reviewer: nit:
```suggestion
The cursor changes to the hand icon. The outline of the switch and thumb should also darken. This helps reinforce that the area is interactive.
```

### Keyboard

> Since the toggle is an interactive component, it must be focusable and keyboard accessible.
> The expected keyboard shortcut for activating a toggle is the Spacebar.

Reviewer: nit:
```suggestion
Since the switch is an interactive component, it must be focusable and keyboard accessible.
The expected keyboard shortcut for activating a switch is the Spacebar key.
```
